doc/marlowe/tutorials/marlowe-data.rst

```diff
@@ -242,20 +276,24 @@ where the types are defined like this:
 
 .. code:: haskell
 
-   data TransactionInput = TransactionInput
-       { txInterval :: SlotInterval
-       , txInputs   :: [Input] }
+   data TOR = TOR { txOutWarnings :: [TransactionWarning]
+                  , txOutPayments :: [Payment]
+                  , txOutState    :: State
+                  , txOutContract :: Contract }
+               deriving (Eq,Ord,Show,Read)
 
    data TransactionOutput =
-       TransactionOutput
-           { txOutWarnings :: [ReduceWarning]
-           , txOutPayments :: [Payment]
-           , txOutState    :: State
-           , txOutContract :: Contract }
-       | Error TransactionError
+      TransactionOutput TOR
+    | Error TransactionError
+   deriving (Eq,Ord,Show,Read)
+
+   data TransactionInput = TransactionInput
+         { txInterval :: SlotInterval
+         , txInputs   :: [Input] }
+      deriving (Eq,Ord,Show,Read)
 
 The notation used here adds field names to the arguments of the
-constructors, giving selectors for the data as well as making in clearer
+constructors, giving selectors for the data as used above, well as used above, making in clearer
```

Comment:
This is garbled, also I just noticed that: `making in clearer -> making it clearer`